dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/trackedentity/hibernate/HibernateTrackedEntityInstanceStore.java

```diff
@@ -909,6 +984,26 @@ else if ( !params.isOrganisationUnitMode( OrganisationUnitSelectionMode.ALL ) )
         return orgUnits.toString();
     }
 
+    /**
+     * Generates an INNER JOIN for program instances. If the param we need to
+     * order by is enrolledAt, we need to join the program instance table to be
+     * able to select and order by this value
+     *
+     * @param params
+     * @return a SQL INNER JOIN for program instances
+     */
+    private String getFromSubQueryJoinProgramInstanceConditions( TrackedEntityInstanceQueryParams params )
+    {
+
+        if ( params.getOrders() != null
+            && params.getOrders().stream().anyMatch( p -> ENROLLED_AT.getPropName().equalsIgnoreCase( p.getField() ) ) )
```

Comment:
If we are doing something like `ENROLLED_AT.getPropName().equalsIgnoreCase` we could maybe add this to the OrderColumn.

`ENROLLED_AT.isEqualToProperty(String)` or `ENROLLED_AT.isProperty(String)`

or so